src/RileyMaze/userFunction.js

```diff
@@ -0,0 +1,244 @@
+/* exported RMZUserFunction */
+
+const {GObject, Gtk} = imports.gi;
+
+const {Codeview} = imports.codeview;
+
+const FORWARD = 1;
+const UP = 2;
+const DOWN = 3;
+const JUMP = 4;
+const RILEY = {
+    // Keep this in sync with riley class in
+    // hack-toy-apps/com.endlessm.RileyMaze/parameters.js
+    queue: [],
+    forward() {
+        this.queue.push(FORWARD);
+    },
+
+    up() {
+        this.queue.push(UP);
+    },
+
+    down() {
+        this.queue.push(DOWN);
+    },
+
+    jump() {
+        this.queue.push(JUMP);
+    },
+};
+
+const COMMON_SCOPE = {
+    // Keep this in sync with UserScope class in
+    // hack-toy-apps/com.endlessm.RileyMaze/parameters.js
+    width: 1920,
+    height: 1080,
+    data: {},
+    riley: RILEY,
+};
+
+const LEVEL_EDIT_SCOPE = {
+    rileyPosition: 0,
+    goalposition: 2,
+    add(unit, x, y) {
+        if (x < 0 || x > 4)
+            throw new RangeError(`${x} must be between 0 and 4`);
+        if (y < 0 || y > 4)
+            throw new RangeError(`${y} must be between 0 and 4`);
+        if (Number.isNaN(Number(x)))
```

Comment:
These checks should go before the previous checks, or they won't have an effect.